cmake/lint.cmake

```diff
@@ -0,0 +1,35 @@
+if(PROGRAM)
```

Comment:
If we're trying to de-clutter the main `CMakeLists.txt`, can we move all of the `add_custom_target` calls into this file?  It doesn't need to be parameterized.

Reply:
I'd need to create an additional file in that case due to the limitations of `add_custom_target`:
- it can't silence output
- overall very inflexible since it won't allow you to run other commands.

Reply:
ok but this file isn't doing much then. It's just shuffling some text into a switch-case.

Reply:
https://github.com/uncrustify/uncrustify/pull/3697